frontend/src/app/Settings/Facility/FacilityForm.tsx

```diff
@@ -369,35 +367,46 @@ const FacilityForm: React.FC<Props> = (props) => {
             <RequiredMessage />
             <FacilityInformation
               facility={facility}
-              updateFacility={updateFacility}
-              errors={errors}
-              validateField={validateField}
               newOrg={props.newOrg}
+              errors={errors}
+              register={register}
+              formCurrentValues={formCurrentValues}
+              setError={setError}
+              getFieldState={getFieldState}
             />
           </div>
         </div>
         <OrderingProviderSettings
-          facility={facility}
-          updateProvider={updateProvider}
-          errors={errors}
-          validateField={validateField}
           newOrg={props.newOrg}
-        />
-        <ManageDevices
-          deviceTypes={props.deviceTypes}
-          selectedDevices={facility.deviceTypes}
-          updateSelectedDevices={updateSelectedDevices}
           errors={errors}
-          clearError={clearError}
-          newOrg={props.newOrg}
+          register={register}
+          formCurrentValues={formCurrentValues}
+        />
+        <Controller
+          render={({ field: { name, ref } }) => (
+            <ManageDevices
+              deviceTypes={props.deviceTypes}
+              newOrg={props.newOrg}
+              errors={errors}
+              formCurrentValues={formCurrentValues}
+              registrationProps={{
+                inputTextRef: ref,
+                setFocus: () => setFocus(name),
+              }}
+              onChange={updateSelectedDevices}
+            />
+          )}
+          defaultValue={formCurrentValues.devices}
+          name="devices"
+          control={control}
+          rules={{ required: "There must be at least one device" }}
```

Comment:
similar note to the comment [here](https://github.com/CDCgov/prime-simplereport/pull/5983/files#r1235287974)